### PR TITLE
gcpdiag: add page

### PR DESCRIPTION
**Pull request by 444B**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):**

### CODE REVIEW

File: pages/common/gcpdiag.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# gcpdiag
+
+> Google Cloud Platform troubleshooting and diagnostics tool.
+> Run in a docker container or in GCP Cloudshell.
+> More information: <https://github.com/GoogleCloudPlatform/gcpdiag>.
+
+- Runs gcpdiag on your project, returning all rules:
```
Comment (kbdharun) [inline pages/common/gcpdiag.md:7]:
```suggestion
- Run `gcpdiag` on your project, returning all rules:
```

File: pages/common/gcpdiag.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+`gcpdiag lint --project={{GCP Project ID}}`
```
Comment (kbdharun) [inline pages/common/gcpdiag.md:9]:
```suggestion
`gcpdiag lint --project={{gcp_project_id}}`
```

File: pages/common/gcpdiag.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+- Runs gcpdiag on your project, hiding rules that are ok:
```
Comment (kbdharun) [inline pages/common/gcpdiag.md:11]:
```suggestion
- Hide rules that are ok:
```
I don't think we need to repeat `runs gcpdiag on your project`.

File: pages/common/gcpdiag.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+`gcpdiag lint --project={{GCP Project ID}} --hide-ok`
+
+- Runs gcpdiag on your project and authenticate using a service account private key file:
+
+`gcpdiag lint --project={{GCP Project ID}} --auth-key {{/path/to/private/key}}`
```
Comment (kbdharun) [inline pages/common/gcpdiag.md:17]:
```suggestion
`gcpdiag lint --project={{gcp_project_id}} --auth-key {{path/to/private_key}}`
```

File: pages/common/gcpdiag.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+> Run in a docker container or in GCP Cloudshell.
+> More information: <https://github.com/GoogleCloudPlatform/gcpdiag>.
+
+- Runs gcpdiag on your project, returning all rules:
+
+`gcpdiag lint --project={{GCP Project ID}}`
+
+- Runs gcpdiag on your project, hiding rules that are ok:
+
+`gcpdiag lint --project={{GCP Project ID}} --hide-ok`
```
Comment (kbdharun) [inline pages/common/gcpdiag.md:13]:
```suggestion
`gcpdiag lint --project={{gcp_project_id}} --hide-ok`
```

File: pages/common/gcpdiag.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+- Runs gcpdiag on your project and authenticate using a service account private key file:
```
Comment (kbdharun) [inline pages/common/gcpdiag.md:15]:
```suggestion
- Authenticate using a service account private key file:
```

File: pages/common/gcpdiag.md (hunk 4 above)
Comment (kbdharun) [inline pages/common/gcpdiag.md:17]:
```suggestion
`gcpdiag lint --project={{gcp_project_id}} --auth-key {{path/to/private_key}}`
```

File: pages/common/gcpdiag.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+`gcpdiag lint --project={{GCP Project ID}} --auth-key {{/path/to/private/key}}`
+
+- Runs gcpdiag on your project, searching logs and metric from {{number}} of days back (default: 3 days):
```
Comment (kbdharun) [inline pages/common/gcpdiag.md:19]:
```suggestion
- Search logs and metrics from a number of days back (default: 3 days):
```

File: pages/common/gcpdiag.md (hunk 4 above)
Comment (kbdharun) [inline pages/common/gcpdiag.md:17]:
```suggestion
`gcpdiag lint --project={{gcp_project_id}} --auth-key {{path/to/private_key}}`
```

File: pages/common/gcpdiag.md (hunk 7 above)
Comment (kbdharun) [inline pages/common/gcpdiag.md:19]:
```suggestion
- Search logs and metrics from a number of days back (default: 3 days):
```

File: pages/common/gcpdiag.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+`gcpdiag lint --project={{GCP Project ID}} --within-days {{number}}`
```
Comment (kbdharun) [inline pages/common/gcpdiag.md:21]:
```suggestion
`gcpdiag lint --project={{gcp_project_id}} --within-days {{number}}`
```

Comment (kbdharun) [inline pages/common/gcpdiag.md:21]:
```suggestion
`gcpdiag lint --project={{gcp_project_id}} --within-days {{number}}`
```

Comment (kbdharun) [inline pages/common/gcpdiag.md:21]:
```suggestion
`gcpdiag lint --project={{gcp_project_id}} --within-days {{number}}`
```

File: pages/common/gcpdiag.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+- Shows the full command options and flags of gcpdiag:
```
Comment (kbdharun) [inline pages/common/gcpdiag.md:23]:
```suggestion
- Display help:
```
This is the generic description we use across all pages.

**Review by kbdharun — COMMENTED** [review]
Thanks for your contribution. I have some suggestions for this page.

**Review by 444B — COMMENTED** [review]
Suggestions accepted, thank you for the formatting guidance @kbdharun

**Review by kbdharun — APPROVED** [review]

File: pages/common/gcpdiag.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+> Run in a docker container or in GCP Cloudshell.
+> More information: <https://github.com/GoogleCloudPlatform/gcpdiag>.
+
+- Run `gcpdiag` on your project, returning all rules:
+
+`gcpdiag lint --project={{gcp_project_id}}`
+
+- Hide rules that are ok:
+
+`gcpdiag lint --project={{gcp_project_id}} --hide-ok`
```
Comment (EmilySeville7cfg) [inline pages/common/gcpdiag.md:13]:
```suggestion
- Hide rules that are ok:

`gcpdiag lint --project="{{id}}" --hide-ok`
```

The same issue as above.

File: pages/common/gcpdiag.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
+# gcpdiag
+
+> Google Cloud Platform troubleshooting and diagnostics tool.
+> Run in a docker container or in GCP Cloudshell.
+> More information: <https://github.com/GoogleCloudPlatform/gcpdiag>.
+
+- Run `gcpdiag` on your project, returning all rules:
+
+`gcpdiag lint --project={{gcp_project_id}}`
```
Comment (EmilySeville7cfg) [inline pages/common/gcpdiag.md:9]:
```suggestion
- Run `gcpdiag` on specific project, returning all rules:

`gcpdiag lint --project="{{id}}"`
```

Optional fix: `specific` may sound better.
Optional fix: placeholder can be simplified.
Optional fix: placeholder can be quoted to protect from misinterpretation by POSIX-compatible shell.

File: pages/common/gcpdiag.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+- Hide rules that are ok:
+
+`gcpdiag lint --project={{gcp_project_id}} --hide-ok`
+
+- Authenticate using a service account private key file:
+
+`gcpdiag lint --project={{gcp_project_id}} --auth-key {{path/to/private_key}}`
```
Comment (EmilySeville7cfg) [inline pages/common/gcpdiag.md:17]:
```suggestion
- Authenticate using a specific service account private key file:

`gcpdiag lint --project="{{id}}" --auth-key {{path/to/private_key}}`
```

Almost the same issue as above.

File: pages/common/gcpdiag.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+- Search logs and metrics from a number of days back (default: 3 days):
+
+`gcpdiag lint --project={{gcp_project_id}} --within-days {{number}}`
```
Comment (EmilySeville7cfg) [inline pages/common/gcpdiag.md:21]:
```suggestion
- Search logs and metrics from a number of days back (default: 3 days):

`gcpdiag lint --project="{{id}}" --within-days {{number}}`
```
Almost the same issue as above.

**Review by EmilySeville7cfg — APPROVED** [review]
No mandatory fixes found, approved.
Thanks for a contribution! 🚀

**Pull request merged**